examples/adaptive_library/BPMTriangularRotorNotches.py

```diff
@@ -0,0 +1,231 @@
+"""
+.. _ref_BPM_Triangular_Rotor_Notches:
+
+Triangular Rotor Notches for IPM
+================================
+Adaptive Template script to create triangular rotor notches to improve NVH performance.
+"""
+# %%
+# This script is designed to be run from Motor-CAD template "e9".
+# If no Motor-CAD file is open,
+# the e9 template will be loaded.
+#
+# This script uses the following adaptive parameters:
+#
+# * Notch Angle (0)
+#
+# * Notch Sweep (2)
+#
+# * Notch Depth (1)
+#
+# * Notches per Pole (2)
+#
+# If these parameters are not already set up in the Motor-CAD file,
+# the parameters will be automatically set,
+# with the default values shown in brackets.
+
+# %%
+# Perform Required imports
+# ------------------------
+# Import pymotorcad to access Motor-CAD.
+# Import Arc, Coordinate, Line, Region and rt_to_xy
+# to define the adaptive template geometry.
+# Import Path, tempfile and shutil
+# to open and save a temporary .mot file if none is open.
+from pathlib import Path
+import shutil
+import tempfile
+
+import ansys.motorcad.core as pymotorcad
+from ansys.motorcad.core.geometry_shapes import triangular_notch
+
+# %%
+# Connect to Motor-CAD
+# --------------------
+# If this script is loaded into the Adaptive Templates file in Motor-CAD,
+# the current Motor-CAD instance will be used.
+# If the script is run externally,
+# and a Motor-CAD instance is currently open,
+# that Motor-CAD instance will be used.
+# If no Motor-CAD instance is open,
+# a new Motor-CAD instance will be opened.
+# TODO: Add explanation that adaptive geometry is only set when this script is loaded in.
+try:
+    # Use existing Motor-CAD instance if possible
+    mc = pymotorcad.MotorCAD(open_new_instance=False)
+except pymotorcad.MotorCADError:
+    # Otherwise open a new instance
+    mc = pymotorcad.MotorCAD()
+    # TODO: What to do if it isn't already open,
+    #  but 'Show GUI when launching Motor-CAD from automation' isn't set?
+
+
+# %%
+# Load file if required
+# ---------------------
+# Check if a file is loaded already.
+# If not, open the e9 IPM motor template,
+# save the file to a temporary folder.
+if mc.get_variable("CurrentMotFilePath_MotorLAB") == "":
+    # Disable popup messages
+    mc.set_variable("MessageDisplayState", 2)
+    mc.set_visible(True)
+    mc.load_template("e9")
+    # TODO: Should we keep Motor-CAD open if we have launched a new instance,
+    #  so the user can see what's been done?
+
+    # Open relevant file
+    working_folder = Path(tempfile.gettempdir()) / "adaptive_library"
+    try:
+        shutil.rmtree(working_folder)
+    except:
+        pass
+
+    Path.mkdir(working_folder)
+    mot_name = "BPMTriRotorNotches"
+    mc.save_to_file(working_folder / (mot_name + ".mot"))
+
+    # Disable adaptive templates
+    mc.set_variable("GeometryTemplateType", 0)
```

Comment:
Worth commenting about *why* this is being done. I assume that this is so the geometry is reset to standard if running the script outside Motor-CAD? If so, this could just be 
```
# Disable adaptive templates so any previous geometry changes
# are reverted, only required when the script is running externally
```

I think that after JackD's changes are implemented (in Motor-CAD and PyMotorCAD), this can be replaced with the reset_adaptive_geometry call?

Reply:
I'm not sure whether line 89 is even necessary. I think if the template has been loaded, this variable will already be set to 0. I may remove this line and the comment.